tests/test_asyncio/test_cache.py

```diff
@@ -228,3 +233,43 @@ async def test_cache_decode_response(self, r):
         assert cache.get(("GET", "foo")) is None
         # get key from redis
         assert await r.get("foo") == "barbar"
+
+
+@pytest.mark.skipif(HIREDIS_AVAILABLE, reason="PythonParser only")
+@pytest.mark.onlynoncluster
+class TestSentinelLocalCache:
+
+    async def test_get_from_cache(self, local_cache, master):
+        await master.set("foo", "bar")
+        # get key from redis and save in local cache
+        assert await master.get("foo") == b"bar"
+        # get key from local cache
+        assert local_cache.get(("GET", "foo")) == b"bar"
+        # change key in redis (cause invalidation)
+        await master.set("foo", "barbar")
+        # send any command to redis (process invalidation in background)
+        await master.ping()
```

Comment:
This step is not needed if you check socket each time before execute command, I think this approach will be more safe and usable from user perspective WDYT?

Reply:
Well, it's not about GET command or any other, the problem is that we need to send an additional read command to server to make sure that we process an invalidation, which leads to data inconsistency in case you're not interacting with server. Another check each time we're executing command add some overhead but it's the only way to make sure that we work with system in correct state, apart from sending additional command before it which also add overhead.

For example:

```
client1 = new Redis(cache=true)
client2 = new Redis(cache=true)

client1.set("key", "value")
client1.get("key") // Executed against server, value is cached
client1.get("key") // Executed against cache

client2.set("key", "new_value") // Client2 updates key tracked by Client1
client1.get("key") // Client1 does not handle invalidation sent by server, because value in cache consider as valid
```

Reply:
You don't _need_ to send an additional command. Invalidations are handled even before a GET call, so in your example the last line will handle the invalidation from the server. Just that the test cases written so far want to emphasize that invalidations are handled on other commands automatically. I will add a test with your scenario, now that we discuss it makes sense to have one.

Reply:
Ah, okay. So it's actually the same what I mean